revokefs/main.c

```diff
@@ -0,0 +1,576 @@
+{
+  if (finfo->fh >= REMOTE_FD_OFFSET)
+    {
+      return request_close (writer_socket, finfo->fh - REMOTE_FD_OFFSET);
+    }
+  else
+    {
+      (void) close (finfo->fh);
+      return 0;
+    }
+}
+
+static int
+callback_fsync (const char *path, int crap, struct fuse_file_info *finfo)
+{
+  if (finfo->fh >= REMOTE_FD_OFFSET)
+    {
+      return request_fsync (writer_socket, finfo->fh - REMOTE_FD_OFFSET);
+    }
+  else
+    {
+      if (fsync (finfo->fh) == -1)
+        return -errno;
+      return 0;
+    }
+}
+
+static int
+callback_access (const char *path, int mode)
+{
+  path = ENSURE_RELPATH (path);
+
+  /* Apparently at least GNU coreutils rm calls `faccessat(W_OK)`
+   * before trying to do an unlink.  So...we'll just lie about
+   * writable access here.
+   */
+  if (faccessat (basefd, path, mode, AT_SYMLINK_NOFOLLOW) == -1)
+    return -errno;
+  return 0;
+}
+
+static int
+callback_setxattr (const char *path, const char *name, const char *value,
+                   size_t size, int flags)
+{
+  return -ENOTSUP;
+}
+
+static int
+callback_getxattr (const char *path, const char *name, char *value,
+                   size_t size)
+{
+  return -ENOTSUP;
+}
+
+/*
+ * List the supported extended attributes.
+ */
+static int
+callback_listxattr (const char *path, char *list, size_t size)
+{
+  return -ENOTSUP;
+
+}
+
+/*
+ * Remove an extended attribute.
+ */
+static int
+callback_removexattr (const char *path, const char *name)
+{
+  return -ENOTSUP;
+
+}
+
+struct fuse_operations callback_oper = {
+  .getattr = callback_getattr,
+  .readlink = callback_readlink,
+  .readdir = callback_readdir,
+  .mknod = callback_mknod,
+  .mkdir = callback_mkdir,
+  .symlink = callback_symlink,
+  .unlink = callback_unlink,
+  .rmdir = callback_rmdir,
+  .rename = callback_rename,
+  .link = callback_link,
+  .chmod = callback_chmod,
+  .chown = callback_chown,
+  .truncate = callback_truncate,
+  .utimens = callback_utimens,
+  .create = callback_create,
+  .open = callback_open,
+  .read = callback_read,
+  .write = callback_write,
+  .statfs = callback_statfs,
+  .release = callback_release,
+  .fsync = callback_fsync,
+  .access = callback_access,
+
+  /* Extended attributes support for userland interaction */
+  .setxattr = callback_setxattr,
+  .getxattr = callback_getxattr,
+  .listxattr = callback_listxattr,
+  .removexattr = callback_removexattr
+};
+
+enum {
+  KEY_HELP,
+};
+
+static void
+usage (const char *progname)
+{
+  fprintf (stdout,
+           "usage: %s basepath mountpoint [options]\n"
+           "\n"
+           "   Makes basepath visible at mountpoint such that files are read-only, directories are writable\n"
+           "\n"
+           "general options:\n"
+           "   -o opt,[opt...]     mount options\n"
+           "   -h  --help          print help\n"
+           "   --socket=fd         Pass in the socket fd\n"
+           "   --exit-with-fd=fd   Pass the read end of a pipe to monitor parent process's exit\n"
```

Comment:
```suggestion
           "   --exit-with-fd=fd   With --backend, exit when the given file descriptor is closed\n"
```

and move this below the help for `--backend`